hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -947,12 +948,13 @@ private Fetch createCircularBiDirectionalFetch(
 			hasBidirectionalFetchParent = true;
 		}
 		else if ( CollectionPart.Nature.fromNameExact( parentNavigablePath.getLocalName() ) != null ) {
-			referencedNavigablePath = parentNavigablePath.getParent().getParent();
+			referencedNavigablePath = getReferencedNavigablePath( creationState, parentNavigablePath );
```

Comment:
Will there be test failures if you use this?

```suggestion
			referencedNavigablePath = getReferencedNavigablePath( creationState, parentNavigablePath.getParent().getParent() );
```

Reply:
yes this change causes failures

Reply:
probably it should be
```
 getReferencedNavigablePath( creationState, parentNavigablePath.getParent())
```